code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -30,22 +30,49 @@
 /datum/ai_behavior/spiderling
 	target_distance = 1
 	base_action = ESCORTING_ATOM
+	var/default_escorted_atom
 
 /datum/ai_behavior/spiderling/New(loc, parent_to_assign, escorted_atom, can_heal = FALSE)
 	. = ..()
+	default_escorted_atom = escorted_atom
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_LIVING, .proc/go_to_target)
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_OBJ, .proc/go_to_obj_target)
 	RegisterSignal(escorted_atom, COMSIG_MOB_DEATH, .proc/spiderling_rage)
 	RegisterSignal(escorted_atom, COMSIG_LIVING_DO_RESIST, .proc/parent_resist)
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/apply_spiderling_jelly)
 	RegisterSignal(escorted_atom, list(COMSIG_XENOMORPH_REST, COMSIG_XENOMORPH_UNREST), .proc/toggle_rest)
+	RegisterSignal(escorted_atom, COMSIG_SPIDERLING_MARK, .proc/decide_mark)
+
+/datum/ai_behavior/spiderling/proc/decide_mark(source, atom/A)
+	SIGNAL_HANDLER
+	if(!A)
+		escorted_atom = source
+		return
+	if(ishuman(A))
+		escorted_atom = null
+		INVOKE_ASYNC(src, .proc/triggered_spiderling_rage, source, A)
+		return
+	if(isobj(A))
+		var/obj/obj_target = A
+		RegisterSignal(obj_target, COMSIG_PARENT_QDELETING, .proc/only_set_escorted_atom)
+		escorted_atom = null
+		INVOKE_ASYNC(src, .proc/obj_mark, source, A)
+		return
+
+/datum/ai_behavior/spiderling/proc/only_set_escorted_atom(source, atom/A)
+	SIGNAL_HANDLER
+	escorted_atom = default_escorted_atom
 
 /// Signal handler to apply resin jelly to the spiderling whenever widow gets it
 /datum/ai_behavior/spiderling/proc/apply_spiderling_jelly()
 	SIGNAL_HANDLER
 	var/mob/living/carbon/xenomorph/spiderling/beno_to_coat = mob_parent
 	beno_to_coat.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
 
+/datum/ai_behavior/spiderling/proc/obj_mark(source, obj/target)
+	SIGNAL_HANDLER
+	go_to_obj_target(source, target)
```

Comment:
A) SIGNAL_HANDLER despite not being a signal handler
B) why this wrapper just call go_to_obj_target directly